develop/how-to-submit-patches/index.md

```diff
@@ -0,0 +1,56 @@
+
+While we prefer merging pull requests, we still accept plain patches too. For those, please use the manual approach described here:
+
+ * **Use the latest version**: Please try to always make your patches against the latest version of wxWidgets.
+
+  In most cases, you should make a patch against the git master branch. If the bug is only in stable branch or it was already fixed in master, make the patch against the stable git branch. You can learn how to download the source from git [here](/develop/code-repository/).
+
+  If you cannot access git (for example when you are behind a firewall), please make the patch against latest release or, better, against fresh git snapshot (latest download: [master.zip](https://github.com/wxWidgets/wxWidgets/archive/master.zip)). 
+
+ * **Standard patch format**: Please do **NOT** send us ZIP archives, whole files or even worse, only code snippets without context. It should be possible to apply your patch using a single `patch -p1 < your.diff` or `patch -p0 < your.diff` command.
+
+  The advantage of using diff is that it produces one file with differences in all files you modified and what's more, diff files are small, easy to read and understand and can be applied even if the affected files have been changed since the moment when the patch was submitted. The _only_ exception to this rule is for the translations: unfortunately, `.po` files change a lot each time they are regenerated because all line numbers recorded in them change, and this risks making any patches to them unappliable very quickly. So for these files, and only for them, please submit the whole files and not patches to the existing ones. You can use `diff` program which is a standard part of most Unix systems and is available as part of cygwin package or elsewhere for Windows or, alternatively, you can just use `git diff` command which works almost in the same manner as diff if you are already using a git clone. The best way to make a patch is to use this command from the root wxWidgets directory:
+
+	git diff origin/master... > mypatch.patch
+
+  Alternatively, use the appropriate base git ref (i.e. for WX_3_0_BRANCH)
+
+  If you use a standalone `diff` you need to add the `-u` option, i.e.:
+
+	diff -uNr wxWidgets-orig wxWidgets-mine > mypatch.patch
+
+  If your patch adds or removes files and you use a standalone diff program locally, you need to use `-N` option for the new files to be included in the patch.
+
+ * Use standard **extension .diff or .patch** for the patch file.
+
+ * **Don't include auto-generated files** (wx/{msw,univ}/setup.h, Makefiles, etc.) in the patch. 
```

Comment:
If we do not include regenerated makefiles, some CI builds may fail?

Reply:
This advice is only for the patches sent by email, not for PRs. Maybe it should be emphasized more above...